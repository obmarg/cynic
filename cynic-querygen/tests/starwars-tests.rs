use insta::assert_snapshot;

use cynic_querygen::{document_to_fragment_structs, QueryGenOptions};

macro_rules! test_query_file {
    ($name:ident, $filename:literal) => {
        #[test]
        fn $name() {
            let schema = include_str!("../../schemas/starwars.schema.graphql");
            let query = include_str!(concat!("queries/starwars/", $filename));

            assert_snapshot!(document_to_fragment_structs(
                query,
                schema,
                &QueryGenOptions::default()
            )
            .expect("QueryGen Failed"))
        }
    };
}

test_query_file!(sanity_test_starwars_query, "sanity.graphql");
test_query_file!(test_nested_arguments, "nested-arguments.graphql");
test_query_file!(bare_selection_sets, "bare-selection-set.graphql");
test_query_file!(multiple_queries, "multiple-queries.graphql");
test_query_file!(fragment_spreads, "fragment-spreads.graphql");
