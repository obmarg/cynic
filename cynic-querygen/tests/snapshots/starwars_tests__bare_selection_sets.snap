---
source: cynic-querygen/tests/starwars-tests.rs
expression: "document_to_fragment_structs(query, schema,\n                             &QueryGenOptions::default()).expect(\"QueryGen Failed\")"
---
#[cynic::query_module(
    schema_path = r#"schema.graphql"#,
    query_module = "schema",
)]
mod queries {
    use super::schema;

    #[derive(cynic::QueryFragment, Debug)]
    #[cynic(graphql_type = "Root")]
    pub struct UnnamedQuery {
        pub all_films: Option<FilmsConnection>,
    }

    #[derive(cynic::QueryFragment, Debug)]
    pub struct FilmsConnection {
        pub films: Option<Vec<Option<Film>>>,
    }

    #[derive(cynic::QueryFragment, Debug)]
    pub struct Film {
        pub id: cynic::Id,
        pub title: Option<String>,
    }

}

mod schema {
    cynic::use_schema!(r#"schema.graphql"#);
}


